### ISSUE TITLE
How to distinguish an object is "large"

### ISSUE THREAD

**Issue by HenryLeongStat** [vatlab/sos-r#12]
As we discussed before, some large objects should be transferred by I/O on disk instead of using memory. However, how can we distinguish the threshold?

1. "Largeness" is different from different machines 
2. Does "largeness" mean number of elements in an `object`? Or size of a `object` saving as file?

`data.frame`, `list`, `matrix` etc. are transferred by I/O on disk (i.e. using `feather`). I suppose the only large objects not using `feather` to be transferred is `vector`, `string`?

**Comment by BoPeng** [comment]
String cannot be large because no one holds book level string in a variable. Vector is the only concern because it can appear from numeric computations. 

The more difficult case is actually dictionaries because it can hold large items.

**Comment by HenryLeongStat** [comment]
Let me see what's the limitation in my 8GB ram laptop. 😄

**Comment by BoPeng** [comment]
Yes, that is something we should do, namely stress testing the magic and see when it breaks. Basically you can use generate larger and larger arrays and see how well they can be passed around.

**Comment by HenryLeongStat** [comment]
And even though I shut down jupyter, those `R` processes are still there and taking all the memory.
